lib/web_ui/test/paragraph_test.dart

```diff
@@ -630,6 +630,204 @@ void main() async {
     );
   });
 
+  testEachMeasurement('getBoxesForRange with maxLines', () {
+    final ParagraphBuilder builder = ParagraphBuilder(ParagraphStyle(
+      fontFamily: 'Ahem',
+      fontStyle: FontStyle.normal,
+      fontWeight: FontWeight.normal,
+      fontSize: 10,
+      textDirection: TextDirection.ltr,
+      maxLines: 2,
+    ));
+    builder.addText('abcd\n');
+    builder.addText('abcdefg\n');
+    builder.addText('ab');
+    final Paragraph paragraph = builder.build();
+    paragraph.layout(const ParagraphConstraints(width: 100));
+
+    // In the dom-based measurement (except Firefox), there will be some
+    // discrepancies around line ends.
+    final isDiscrepancyExpected =
```

Comment:
It seems the discrepancy is always an empty box at the end of the line. Can we inject them ourselves and remove the discrepancy?

Reply:
To be clear, that's a discrepancy between Flutter and Chrome. Flutter doesn't include an empty box in the end of line, but Chrome does. (Firefox matches Flutter's behavior here).

Injecting an empty box would remove the discrepancy, you are right. But we would be deviating from Flutter.

Reply:
I'll see if I can figure out a way to identify and remove the empty boxes that Chrome returns.

Reply:
I see. Yeah, in that case what about the reverse, remove the empties?

Reply:
Done.